gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlSpecStoreTest.java

```diff
@@ -215,6 +246,60 @@ public void testGetSpec() throws Exception {
     Assert.assertEquals(specs.size(), 1);
     Assert.assertTrue(specs.contains(this.flowSpec4));
   }
+  @Test
+  public void testGetSpecsAfterUpdate() throws Exception {
+    this.specStore.addSpec(this.flowSpec5);
+    FlowSpec result = (FlowSpec) this.specStore.getSpec(this.uri5);
+    removeModificationTimestampFromSpecs(result);
+    Assert.assertEquals(result, this.flowSpec5);
+    // Updating  flowSpec5
+    this.specStore.addSpec(this.flowSpec5Updated);
```

Comment:
this update comes too soon, prior to thoroughly validating the "before state".

ideally both before and after would validate the same search param sets.  the majority would demonstrate "flip-flop", wherein what retrieved a result before no longer does post-update or vice-versa.  a small subset would work (or not) on both to verify intentionally unchanged/uncorrupted search facets.

one option is for the "before" validations to live in a separate method or even a heavily augmented `testGetSpec` (the one just above).  just use the same starting flowSpec there.

personally, I'd define the same `FlowSpecSearchObject` instances common to both and then use them while initializing two lists of input pairs, which I'd loop over to run through, once before update and then again after -
```
// (reappropriate `Map.Entry` as a "Pair")
List<Pair<Map.Entry<FlowSpecSearchObject, Boolean>> preUpdateValidations = ...
List<Pair<Map.Entry<FlowSpecSearchObject, Boolean>> postUpdateValidations = ...

Reply:
addressed